linkschkr_test.go

```diff
@@ -0,0 +1,69 @@
+package linkschkr
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestCrawler(t *testing.T) {
+	t.Parallel()
+
+	nWorkers := 1000 // just for fun
+	work := make(chan *Work)
+	results := make(chan *Result)
+
+	for n := 0; n < nWorkers; n++ {
+		go Crawler(n, work)
+	}
+
+	body := `<html>
+	<head>
+		<title>For the love of Go!</title>
+	</head>
+	<body>
+		Hi folks,
+
+		See docs <a href="/docs">Go docs</a>. Go is awesome!
+		ajkwerjakwerjkaewrhjakwehrjkawe
+		aelwrkhakelrhalkewrhkleawlhrwaklerhalkewrhalewrk
+
+		Visit <a href="/live">our live</a>.
+		aewrjaewporawe
+
+		awerjaioewrhaewoi
+		a href="testintentional.com">trying to get you in trouble</a>
+	</body>
+</html>`
+
+	ts := httptest.NewServer(
+		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if r.URL.Path == "/" {
+				w.Header().Add("Content-Type", "application/text")
+				w.Write([]byte(body))
```

Comment:
The easiest way of getting a string into an `io.Writer` is to print it to the writer with `fmt.Fprintf`. But if you put the HTML into a file, you will have an `io.Reader` (perhaps `f`), so the easiest thing will be to copy the whole reader to the writer:

```go
io.Copy(w, f)
```